go/common/future/future.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2025 Sonic Operations Ltd
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at soniclabs.com/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package future
+
+// Futures provided a simple abstraction for asynchronous computation results.
```

Comment:
This is the package level doc, right?
```suggestion
// Future provides a simple abstraction for asynchronous computation results.
```